src/ssort/_utils.py

```diff
@@ -40,3 +41,22 @@ def sort_key_from_iter(values):
     index = {statement: index for index, statement in enumerate(values)}
     key = lambda value: index[value]
     return key
+
+
+_ENCODING_RE = re.compile("^[ \t\f]*#.*?coding[:=][ \t]*([-_.a-zA-Z0-9]+)")
+
+
+def detect_encoding(bytestring):
+    """
+    Detect the encoding of a python source file based on "coding" comments, as
+    defined in [PEP 263](https://www.python.org/dev/peps/pep-0263/).
+
+    Defaults to "utf-8", as per [PEP 3120](https://www.python.org/dev/peps/pep-3120/).
+    """
+    encoding = "utf-8"
+    for line in bytestring.splitlines()[:2]:
+        match = _ENCODING_RE.match(line.decode("latin-1"))
+        if not match:
+            continue
+        encoding = match[1]
+    return encoding
```

Comment:
Indeed, yes.  It would appear that the python interpreter will always go with the first encoding comment.

While trying to figure out exactly what the python interpreter does I stumbled across [tokenize.detect_encoding](https://docs.python.org/3/library/tokenize.html#tokenize.detect_encoding), which appears to do exactly what we want.  Should be able to just drop it in.